editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -3431,16 +3431,18 @@ void CanvasItemEditor::_draw_invisible_nodes_positions(Node *p_node, const Trans
 		_draw_invisible_nodes_positions(p_node->get_child(i), parent_xform, canvas_xform);
 	}
 
-	if (canvas_item && !canvas_item->_edit_use_rect() && (!editor_selection->is_selected(canvas_item) || _is_node_locked(canvas_item))) {
-		Transform2D xform = transform * canvas_xform * parent_xform;
-
-		// Draw the node's position
-		Ref<Texture> position_icon = get_icon("EditorPositionUnselected", "EditorIcons");
-		Transform2D unscaled_transform = (xform * canvas_item->get_transform().affine_inverse() * canvas_item->_edit_get_transform()).orthonormalized();
-		Transform2D simple_xform = viewport->get_transform() * unscaled_transform;
-		viewport->draw_set_transform_matrix(simple_xform);
-		viewport->draw_texture(position_icon, -position_icon->get_size() / 2, Color(1.0, 1.0, 1.0, 0.5));
-		viewport->draw_set_transform_matrix(viewport->get_transform());
+	if (show_2d_gizmos) {
```

Comment:
You could have moved the condition inside the next if, that would have avoid nested ifs which make things harder to read IMHO.

Reply:
Yes, I can agree with you. However these two conditions are concerning separate logic bits, that's why I didn't glue them together. 
Also long `if` statements with mixed flags are not improving the readability either, making the person who reads them to scroll IDE window trying to parse-and-memorize things, often dealing with totally different behavior flags. 
Do we have something in official codestyle regarding code areas like that?

Reply:
Not really for things like that I guess. 
I don't care that much to be honest, I tend to limit the number of nested structure when I can, since it does take more space and is harder to read in my opinion (especially in the canvas item editor, which is quite a complex file). But I understand your point, it is valid too, so you can keep it that way I believe.